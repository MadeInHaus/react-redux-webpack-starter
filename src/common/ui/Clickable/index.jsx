import React from 'react';
import PropTypes from 'prop-types';
import { Link } from '@ui';
import cx from 'classnames';

import styles from './Clickable.scss';

const Clickable = ({ className, href, theme, to, ...props }) => {
    const classNames = cx(styles.root, className, styles[theme]);

    if (to || href) {
        return (
            <Link {...props} className={classNames} to={to} href={href}>
                <span>{props.children}</span>
            </Link>
        );
    }
    return (
        <button {...props} className={classNames}>
            <span>{props.children}</span>
        </button>
    );
};

Clickable.propTypes = {
    children: PropTypes.Node,
    className: PropTypes.string,
    href: PropTypes.string,
    theme: PropTypes.oneOf(Object.keys(styles)),
    to: PropTypes.string,
};

Clickable.defaultProps = {
    className: null,
    href: null,
    theme: 'primary',
    to: null,
};

export default Clickable;
