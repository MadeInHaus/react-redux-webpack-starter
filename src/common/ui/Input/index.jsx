import React from 'react';
import { string } from 'prop-types';
import cx from 'classnames';

import { Clickable } from '@ui';

import styles from './Input.scss';

const Input = ({ type, ...props }) => {
    switch (type) {
        case 'select':
            return (
                <select className={cx(styles.root, styles.select)} {...props} />
            );
        case 'textarea':
            return (
                <textarea
                    className={cx(styles.root, styles.textarea)}
                    {...props}
                />
            );
        case 'submit':
        case 'button':
            return <Clickable theme="primary" {...props} />;
        default:
            return <input type={type} className={styles.root} {...props} />;
    }
};

Input.propTypes = {
    type: string,
};

Input.defaultProps = {
    type: 'text',
};

export default Input;
