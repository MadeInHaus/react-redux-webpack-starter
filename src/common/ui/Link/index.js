import React from 'react';
import { string } from 'prop-types';
import { Link as RouterLink, NavLink } from 'react-router-dom';

const Link = ({ to, activeClassName, ...props }) => {
    if (to) {
        return activeClassName ? (
            <NavLink activeClassName={activeClassName} to={to} {...props} />
        ) : (
            <RouterLink to={to} {...props} />
        );
    }

    return <a {...props} />;
};

Link.propTypes = {
    activeClassName: string,
    to: string,
};

export default Link;
