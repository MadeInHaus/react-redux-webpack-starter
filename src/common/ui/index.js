export { default as Clickable } from './Clickable';
export { default as Img } from './Img';
export { default as ImgPreload } from './Img/ImgPreload';
export { default as Input } from './Input';
export { default as Link } from './Link';
export { default as Loader } from './Loader';
export { default as SvgIcon } from './SvgIcon';
export { default as Text } from './Text';
