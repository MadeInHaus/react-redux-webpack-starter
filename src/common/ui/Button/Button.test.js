import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from 'ui';

test('When Button is Clicked', () => {
    const component = renderer.create(<Button>Facebook</Button>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
