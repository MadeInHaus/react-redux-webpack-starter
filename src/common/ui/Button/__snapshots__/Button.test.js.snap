// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Button is Clicked 1`] = `
<button
  className="root body"
>
  Facebook
</button>
`;
